[DMN] feat(zammad) add zammad

Emalton opened the pull request:
**Description**
<!--
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
-->
Add Zammad to incubator. Discord thread link is [here](https://discordapp.com/channels/830763548678291466/1152627719176192081/1152627722380652545) to track progress and discuss implementation. Note: Elasticsearch isn't in the core/dependency lib yet.

⚒️ Fixes  # <!--(issue)-->

**⚙️ Type of change**

- [X] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->
This is very much untested, not sure how dependencies work with the environment variables so will likely need help with that.

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [X] ⚖️ My code follows the style guidelines of this project
- [ ] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [X] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [X] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [X] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

Emalton committed:
Add Zammad chart to incubator

Very much untested, not sure how dependencies work with parameters. Needs to be reviewed.

StevenMcElligott commented on charts/incubator/zammad/values.yaml:
```diff
@@ -0,0 +1,80 @@
+  rails:
+    rails_trustedproxies: ['127.0.0.1', '::1']
+  redis:
+    redis_url: "http://redis.ix-redis.svc.cluster.local:6379"
```
```suggestion
redis_url:
              secretKeyRef:
                expandObjectName: false
                name: '{{ printf "%s-%s" .Release.Name "rediscreds" }}'
                key: url
```

xstar97 replied:
why are we adding the redis dependency here???

Emalton replied:
Good call, I originally had them as config values but I believe they can be moved to right to the workload, yeah? Like below.
```yml
REDIS_URL:
  secretKeyRef:
    expandObjectName: false
    name: '{{ printf "%s-%s" .Release.Name "rediscreds" }}'
    key: url

Emalton replied:
Yep they can - just saw your reference [here](https://github.com/truecharts/charts/blob/7c3adc77d082b257a10afe839905ef8ae6af1b52/charts/stable/rsshub/values.yaml#L49). Should be all set with this one!

StevenMcElligott commented on charts/incubator/zammad/values.yaml:
```diff
@@ -0,0 +1,80 @@
+  redis:
+    redis_url: "http://redis.ix-redis.svc.cluster.local:6379"
+  memcached:
+    memcached_url: "http://memcached.ix-memcached.svc.cluster.local:11211"
```
```memcached:
   memcached_url: "{{ printf "%v-%v:%v" .Release.Name "memcached" "11211" }}"```

Emalton replied:
Got it, but had to use `'` instead of `"` for the enclosing quotes. (Shout out to stavros for showing me that playground website which helped!)

StevenMcElligott commented on charts/incubator/zammad/values.yaml:
```diff
@@ -0,0 +1,80 @@
+    podSpec:
+      containers:
+        main:
+          probes:
```
Why are the probes disabled btw?

Emalton replied:
That sir is an excellent question - probably because I duplicated an existing chart that had them disabled and didn't enable them! I went ahead enabled all three of them.

StevenMcElligott commented on charts/incubator/zammad/values.yaml:
```diff
@@ -0,0 +1,80 @@
+    webcon_schedprocesses: false
+    webcon_delayedjobs: 0
+  rails:
+    rails_trustedproxies: ['127.0.0.1', '::1']
```
Your lint is failing here btw

xstar97 replied:
this is how its done... use [] for empty list.
```yaml
rails_trustedproxies:
  - 127.0.0.1
```

Emalton replied:
Got it, added the IPv4 and IPv6 addresses in that list in both locations. Thanks guys!

xstar97 commented on charts/incubator/zammad/questions.yaml:
```diff
@@ -0,0 +1,200 @@
+                description: Allows you to store your application cache files within Memcached, and should point to your Memcached instance. The default value assumes your Memchaced chart's name is memcached. If not provided, Zammad falls back to the file system (`/opt/zammad/tmp/cache*`).
+                schema:
+                  type: string
+                  default: "http://memcached.ix-memcached.svc.cluster.local:11211"
```
whats the deal with this?

StevenMcElligott replied:
Yep not necessary in the questions at all TBH

Emalton replied:
Fixing and will commit shortly. How should elasticsearch be handle in regards to the params such as the SHEMA/PORT?
![image](https://github.com/truecharts/charts/assets/9328458/6549e2f2-986c-4697-9f33-08e359bcd463)

Emalton replied:
Marking as resolved, answered in stavros' review below.

Emalton committed:
Update charts/incubator/zammad/values.yaml

Commit Steven's proposed changes

Co-authored-by: StevenMcElligott <89483932+StevenMcElligott@users.noreply.github.com>
Signed-off-by: John P <emaltonator@gmail.com>

Emalton committed:
Add redis dependency

Add the redis dependency that I forgot to commit

Emalton committed:
Redis Updates

- Removed Redis config from the questions.yaml file
- Move Redis URL/Secret from the question variables section to the environment variables section.

Emalton committed:
Fixed lists & enable probes

- Enabled the probes.
- Fixed the list of addresses for the trusted rails proxies.

Emalton committed:
Elasticsearch Updates

- Removed Elasticsearch URL config from the questions.yaml file
- Move Elasticsearch URL from the question variables section to the environment variables section.

Emalton committed:
Memcached Updates

- Removed Memcached URL config from the questions.yaml file
- Move Memcached URL from the question variables section to the environment variables section.

Emalton committed:
IP Address Fix

Surround IP addresses in quotations in an attempt to fix a lint error.

Emalton committed:
Trusted Proxies Fix?

Attempt to fix the trusted proxies by converting to JSON for the environment variable.

Emalton committed:
Revert: Fix IP Addresses

Revert the attempted fix for IP addresses.

Emalton committed:
Fixed template logic to correct question

Fixed the template logic for the trusted proxies so it references the proper question.

stavros-k commented on charts/incubator/zammad/values.yaml:
```diff
@@ -0,0 +1,94 @@
+    elasticsearch_schema: "http"
+    elasticsearch_namespace: "zammad"
+    elasticsearch_reindex: true
+    elasticsearch_sslverify: true
```
Why would those be exposed if the ES is used as a dependency? (meaning that no configuration is needed from the user)

Emalton replied:
Good call, will fix this and update the documentation.

stavros-k commented on charts/incubator/zammad/docs/installation_notes.md:
```diff
@@ -0,0 +1,43 @@
+
+- The `Elasticsearch Reindex` parameter is for enabling reindexing on startups, which may be troublesome on larger installations. Disabling this setting requires you to re-index your search index manually whenever that's needed by upgrading to a new Zammad version! This is enabled by default.
+
+- The `Elasticsearch SSL Verify` parameter allows you to let the compose scripts ignore self signed SSL certificates for your Elasticsearch installation if needed. This is enabled by default.
```
Dependency configuration is not meant to be user configurable

Emalton replied:
Including `Elasticsearch Reindex` parameter? It's for enabling reindexing on startups, which Zammad says can be problematic on larger installs. Even though this is for a dependency, should this still be user configurable? I feel like it should be so this tune options are available to the user. What are your thoughts?

stavros-k commented on charts/incubator/zammad/questions.yaml:
```diff
@@ -0,0 +1,172 @@
+                description: Allows you to specify the amount of processes that should work on delayed jobs. Increasing this can improve issues with delayed jobs stacking up in your system. You may want to try to use `Concurrent Session Jobs` before though. This option can be very CPU intensive. The maximum value is 16.
+                schema:
+                  type: int
+                  default: 0
```
Are any of those values are required? eg I can leave it blank?

Emalton replied:
Good call, I addressed this in latest commit. The integers and booleans should be required, and the string for web concurrency (instances) isn't required.

stavros-k commented on charts/incubator/zammad/questions.yaml:
```diff
@@ -0,0 +1,172 @@
+                label: Elasticsearch SSL Verify
+                description: Allows you to let the compose scripts ignore self signed SSL certificates for your Elasticsearch installation if needed.
+                schema:
+                  type: bool
```
`boolean`

But this section should not be here anyway

Emalton committed:
Remove dependency questions

Removed dependency questions that shouldn't be there as they are fixed and don't need to be user configurable per stavros' suggestions.

Emalton committed:
Question requirements

Set question requirements
